src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -753,13 +753,46 @@ public void stateChanged(ChangeEvent e)   {
         }
     }
 
-    // PENDING(hmuller) - the next three methods should be removed
+    /**
+     * Unlike most components, JScrollBar will derive the minimum size from
+     * the preferred size in one axis and a fixed minimum size in the other
+     * unless otherwise set by the user.
+     * Thus, it overrides {@code JComponent.setMinimumSize} contract
+     * that subsequent calls to getMinimumSize will return the
+     * same value as set in {@code JComponent.setMinimumSize}
```

Comment:
Doesn't the javadoc contradict the implementation?

This fix has restored the contract of `setMinimumSize` / `getMinimumSize`, so `JScrollBar` is now like other components.

What we can add though is how the size is calculated when it's not set, however, it's documented in the `get-` methods.